content/4_Gold/Meet_In_The_Middle.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: meet-in-the-middle
+title: 'Meet In The Middle'
+author: Chongtian Ma
+description:
+  'Divide the search space into two'
+frequency: 1
+prerequisites:
+---
+
+## Meet In The Middle
+
+<FocusProblem problem="sample" />
+
+
+### Tutorial
+
+<Resources>
+	<Resource source="CPH" title="5.5 - Meet in the Middle" />
+	<Resource
+		source="Errichto"
+		title="Meet in the Middle | Tutorial & Problems"
+		url="https://www.youtube.com/watch?v=18sJ3mK173s"
+		starred= "true"
+	/>
+</Resources>
+
+<Info title="Pro Tip">
+
+Meet in the middle problems often have double the constraints of an already optimized solution.
+
+</Info>
+
+### Naive Solution
+
+Loop through all subsets in the array and if the sum is equal to $x$, then increase our answer. Worst case this does about $4^{40}$ operations, which is way too slow.
+
+### Meet in the Middle Solution
+
+We can divide the given array into two separate arrays. Let's say that the $left$ array runs from indexes $0$ to $n/2-1$, and the $right$ array runs from indexes $n/2$ to $n-1$. Both arrays will have at most $20$ elements, so we can loop through all subsets of these two arrays in at most $2^{21}$ operations, which is perfectly fine.
+
+Now that we've got the subset sums of these two separate arrays, we need to recombine them to search for our answer. For every sum in the $left$, we can simply check how many elements of $x - $sum there are in $right$. This can be done using simple binary search.
```

Comment:
```suggestion
Now that we've got the subset sums of these two separate arrays, we need to recombine them to search for our answer. For every $\texttt{sum}$ in the $left$, we can simply check how many elements of $x - \texttt{sum}$ there are in $right$. This can be done using simple binary search.
```